ankaios_sdk/_components/workload_state.py

```diff
@@ -0,0 +1,474 @@
+    "(int): The workload failed due to execution failure."
+    FAILED_UNKNOWN: int = 12
+    "(int): The workload failed due to an unknown reason."
+    FAILED_LOST: int = 13
+    "(int): The workload failed because it was lost."
+    NOT_SCHEDULED: int = 14
+    "(int): The workload is not scheduled."
+    REMOVED: int = 15
+    "(int): The workload has been removed."
+
+    def __str__(self) -> str:
+        """
+        Return the name of the enumeration member.
+
+        Returns:
+            str: The name of the enumeration member.
+        """
+        return self.name
+
+    @staticmethod
+    def _get(state: WorkloadStateEnum,
+             field: _ank_base) -> "WorkloadSubStateEnum":
+        """
+        Get the enumeration member corresponding to the given state and field.
+
+        Args:
+            state (WorkloadStateEnum): The state of the workload.
+            field (_ank_base): The field to look up.
+
+        Returns:
+            WorkloadSubStateEnum: The enumeration member corresponding
+                to the state and field.
+
+        Raises:
+            ValueError: If the field does not correspond to
+                any enumeration member.
+        """
+        # SNAKE_CASE to CamelCase
+        state_name = "".join([elem.title() for elem in state.name.split("_")])
+        if field not in getattr(_ank_base, state_name).values():
+            raise ValueError("No corresponding WorkloadSubStateEnum "
+                             + f"value for enum: {field}")
+        return WorkloadSubStateEnum[getattr(_ank_base, state_name).Name(field)]
+
+    def _sub_state2ank_base(self) -> _ank_base:
+        """
+        Convert the WorkloadSubStateEnum member to the corresponding
+        _ank_base value.
+
+        Returns:
+            _ank_base: The corresponding _ank_base value.
+
+        Raises:
+            ValueError: If there is no corresponding _ank_base
+                value for the enumeration member.
+        """
+        try:
+            return getattr(_ank_base, self.name)
+        except AttributeError as e:  # pragma: no cover
+            raise ValueError("No corresponding ank_base value "
+                             + f"for enum: {self.name}") from e
+
+
+# pylint: disable=too-few-public-methods
+class WorkloadExecutionState:
+    """
+    Represents the execution state of a workload.
+
+    Attributes:
+        state (WorkloadStateEnum): The state of the workload.
+        substate (WorkloadSubStateEnum): The sub-state of the workload.
+        info (str): Additional information about the workload state.
+    """
+    def __init__(self, state: _ank_base.ExecutionState) -> None:
+        """
+        Initializes a WorkloadExecutionState instance.
+
+        Args:
+            state (_ank_base.ExecutionState): The execution state to interpret.
+        """
+        self.state: WorkloadStateEnum = None
+        self.substate: WorkloadSubStateEnum = None
+        self.info: str = None
```

Comment:
Let's call this `additional_info` as in the proto

Reply:
Done.